### PR TITLE
convert back to gamma to avoid multiple toLinear conversions

### PR DESCRIPTION
**Pull request by TrevorDev**
https://github.com/BabylonJS/Babylon.js/issues/4060

Updated playground for testing: http://localhost:1338/Playground/index-local.html#1YAIO7#35
Had to modify texture type to halffloat to avoid banding:
`mirrorMaterial.reflectionTexture = new BABYLON.MirrorTexture("mirror", 1024, scene, true, BABYLON.Engine.TEXTURETYPE_HALF_FLOAT);`

### CODE REVIEW

File: src/Materials/Textures/babylon.mirrorTexture.ts [hunk 1]
```diff
@@ -90,6 +90,7 @@
             this.ignoreCameraViewport = true;
 
             this.onBeforeRenderObservable.add(() => {
+                this.gammaSpace = !scene.imageProcessingConfiguration.isEnabled || !scene.imageProcessingConfiguration.applyByPostProcess;
```
Comment (TrevorDev) [inline src/Materials/Textures/babylon.mirrorTexture.ts]: Double check that this is the correct place to set this? Should there be similar logic added to the refractionTexture?

File: src/Shaders/default.fragment.fx [hunk 2]
```diff
@@ -304,13 +309,17 @@ void main(void) {
...
-#endif
+	#ifdef REFLECTIONMAP_PROJECTION
+		coords /= vReflectionUVW.z;
+	#endif
 
 	coords.y = 1.0 - coords.y;
-
-	reflectionColor = texture2D(reflection2DSampler, coords).rgb * vReflectionInfos.x;
+	#ifdef IS_REFLECTION_LINEAR
+		reflectionColor = toGammaSpace(texture2D(reflection2DSampler, coords).rgb);
```
Comment (TrevorDev) [inline src/Shaders/default.fragment.fx]: is the toGammaSpace also needed on the textureCube lookups above?

File: src/Materials/Textures/babylon.mirrorTexture.ts (hunk 1 above)
Comment (sebavan) [reply src/Materials/Textures/babylon.mirrorTexture.ts]: Could it be done on creation and in the onUpdateParameters event of the imageProcessingConfiguration ???

File: src/Shaders/default.fragment.fx [hunk 3]
```diff
@@ -274,7 +274,12 @@ void main(void) {
 
 	refractionCoords.y = 1.0 - refractionCoords.y;
 
-	refractionColor = texture2D(refraction2DSampler, refractionCoords).rgb * vRefractionInfos.x;
```
Comment (sebavan) [inline src/Shaders/default.fragment.fx]: I guess we should do the same with the cube sampler a bit upper in the file.

**Comment by sebavan** [comment]
adding @deltakosh for review.

**Pull request merged**